### PR TITLE
[RELAY][OP] expand_dims

### PR DESCRIPTION
**Pull request by junrushao1994**
This PR contains a relay operator, `expand_dims`.

TODO:
- [x] Docstrings
- [x] I didn't see a boundary checking for `axis` in [topi](https://github.com/dmlc/tvm/blob/0247b97198b0fecf82e6ca44d2f34bc8dfc3540f/topi/include/topi/transform.h#L42). Did I miss anything?
- [x] API inconsistency with [numpy.expand_dims](https://docs.scipy.org/doc/numpy/reference/generated/numpy.expand_dims.html). Our API looks like `expand_dims(data, axis, num_newaxis=1)`, but numpy API is `expand_dims(a, axis) `. I think it is okay because ours seem to be a superset.
- [x] I implemented `axis` and `num_newaxis` as `int`. Wondering if it is possible to make them `IndexExpr`, so that `axis` could depend on some intermediate result of computation.

I am working on the docstring, and @tqchen could you help comment on the TODO 2, 3, 4?

Related issue: #1799

### CODE REVIEW

**Comment by tqchen** [comment]
- bound checking should be added to topi
- It is intended as a superset and the default case is still numpy consistent
- For now, int is fine

**Comment by tqchen** [comment]
Please also request reviews from reviewers

**Comment by junrushao1994** [comment]
@tqchen @srkreddy1238 Hey could you help review this PR?

**Comment by junrushao1994** [comment]
Rebased to current master.

**Comment by tqchen** [comment]
sorry, still need another rebase after https://github.com/dmlc/tvm/pull/1821

File: python/tvm/relay/op/transform.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+    axis : int
+        The axis at which the input array is expanded.
+        Should lie in range `[-data.ndim - 1, data.ndim]`.
+        If `axis < 0`, it is the first axis inserted;
+        If `axis >= 0`, it is the last axis inserted in Python's negative indexing.
+
+    num_newaxis : int
+        Number of axises to be inserted. Should be >= 0.
+    """
```
Comment (tqchen) [inline python/tvm/relay/op/transform.py]:
```
Returns 
--------
```

**Comment by tqchen** [comment]
Please also add the operator to  https://github.com/dmlc/tvm/blob/master/docs/langref/relay_op.rst

File: python/tvm/relay/op/transform.py (hunk 1 above)
Comment (junrushao1994) [reply python/tvm/relay/op/transform.py]: @tqchen Sorry for my poor English...What does this sentence mean: "The number of dimensions is num_newaxis greater than that of the input."?

Comment (tqchen) [reply python/tvm/relay/op/transform.py]: maybe result_dimension = source dimension  + num_newaxis

Comment (junrushao1994) [reply python/tvm/relay/op/transform.py]: @tqchen Good point! Like this?

```
result : relay.Expr
    The reshaped result. `result.ndim = data.ndim + num_newaxis`.
```

**Pull request merged**